ESH-INF/thing/_channels.xml

```diff
@@ -3,15 +3,31 @@
 	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
 	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
 
-	<!-- Color Channel -->
-	<channel-type id="color_color">
-		<item-type>Color</item-type>
-		<label>Color</label>
-		<description>The color channel allows to control the color of a light.
+    <!-- Color Channel -->
+    <channel-type id="color_color">
+        <item-type>Color</item-type>
+        <label>Color</label>
+        <description>The color channel allows to control the color of a light.
             It is also possible to dim values and switch the light on and off.
         </description>
-		<category>ColorLight</category>
-	</channel-type>
+        <category>ColorLight</category>
+    </channel-type>
+
+    <!-- Color Channel -->
+    <channel-type id="color_mode">
+        <item-type>Number</item-type>
+        <label>Color</label>
+        <description>The color mode channel allows to control the color mode of a light.
+        </description>
+        <category>ColorLight</category>
+        <state readOnly="true">
+            <options>
+                <option value="0">Hue and Saturation</option>
```

Comment:
What happens if you send a command to the color_temperature channel when this mode is selected?

Reply:
It is up to the bulb. Some bulbs I have update the temperature, some don't. Temperature and colour aren't linked are they? What does Hue do?

Reply:
And here's again where you need to build the abstraction: If the user sends color command, it is expected to be executed. It is your job in the binding to make this sure. If you need to read and change the color_mode first, so be it. But this must be hidden in the binding and not be exposed to the user.

Reply:
Yes - again, as stated elsewhere this is what the binding does. It doesn't stop the bulb reporting what mode it is using though does it?

I think you don't understand this - the binding is presenting all the standard features to the user in the standard way - this mode status is on the bulb interface only, so it's indicating what the bulb is doing only.

Reply:
The bulbs isn't "doing" anything - it has a (visible) state, which is some color, which should be reflected as the state of the color channel - that's all the user needs to know.